openedx/features/survey_report/api.py

```diff
@@ -35,3 +37,45 @@ def generate_report() -> None:
     )
 
     survey_report.save()
+    return survey_report.id
+
+def send_report(report_id: int) -> None:
+    """
+    Send a report to Openedx endpoint and save the response in the SurveyReportUpload model.
+
+    endpoint: The value of the setting SURVEY_REPORT_ENDPOINT
+
+    content_type: JSON
+
+    payload:
+    - courses_offered: Total number of active unique courses.
+    - learner: Recently active users with login in some weeks.
+    - registered_learners: Total number of users ever registered in the platform.
+    - enrollments: Total number of active enrollments in the platform.
+    - generated_certificates: Total number of generated certificates.
+    - extra_data: Extra information that will be saved in the report, E.g: site_name, openedx-release.
+    - created_at: Date when the report was generated, this date will send with format '%m-%d-%Y %H:%M:%S'
+    """
+    report = SurveyReport.objects.get(id=report_id)
+
+    fields = [
+        "courses_offered",
+        "learners",
+        "registered_learners",
+        "generated_certificates",
+        "enrollments",
+    ]
+
+    data = model_to_dict(report, fields=fields)
+    data["extra_data"] = report.extra_data
+    data["created_at"] = report.created_at.strftime("%m-%d-%Y %H:%M:%S")
+
+    request = requests.post(settings.SURVEY_REPORT_ENDPOINT, json=data)
```

Comment:
@ormsbee do you have something different in mind for sending this report?

We have explored things like having a zappier that formats and sends to google docs, but we are open to anything.

Reply:
I think we should have a real URL enabled by default (if one exists yet), and a timeout on this request so it doesn't block for too long. But other than that, this seems fine to me.

Reply:
Is there an endpoint to send to at this point?

Reply:
@ormsbee No, we don't have an endpoint for it, @felipemontoya some approach for this?

Reply:
I can produce a Zappier webhook and we leave that as the default for now. 
https://help.zapier.com/hc/en-us/articles/8496326446989

Webhooks are available only in paid plans. I can put it in the edunext account.

Reply:
@e0d, @jmakowski1123: Did you folks have thoughts w.r.t. where this data gets collected, and what the access policies are for that destination?

Reply:
I don't have specific thoughts about what service we should use, from a hat, something like S3 via an API gateway, Zapier, or something similarly generic.  I think encryption in transit is a hard requirement.  

I think instances would have write only permissions.

Access to the raw survey data would be highly restricted.

Reply:
Looking at the sheet and the JSON data structure, I have a question.  The site URL feels like an essential unique key for this data, but it seem to be optional and in the extra_data field.  I would imagine that we would want to capture rows for many sites in a single sheet/table and to be able to differentiated each row by site and timestamp.

Reply:
From the data analysis point of view yes. The site url is very important to differentiate the data.

From the point of view of the site sharing the data [it was proposed that sites maintain](https://docs.google.com/document/d/1tm1O5BQsAwekphI5IN9pBn8behQDZigXePLGqCl5he8/edit?disco=AAAAeHzDJ4k) the option of making the report anonymously.

A compromise could be that we set a unique differentiator for the report, namely the site. However if the site operator wishes to remain anonymous we could hash the site value one-directionally. E.g sha256

Reply:
@felipemontoya @jmakowski1123 @e0d I added the site name in a new field and I added a setting to make the site name anonymous.

new survey report file: https://docs.google.com/spreadsheets/d/1PWhWZ0XN6tEo8xhDh3kTSuGMrEYAE6tDBQu2kdF8tZA/edit?usp=sharing

setting: ANONYMOUS_SURVEY_REPORT is False by default.

Reply:
How is the hash computed?  Ideally it would be a consistent value -- it wouldn't change if the secret key was updated for example.

Reply:
@e0d I'm using a sha256 from the hashlib so the hash will be always the same.

Reply:
At the risk of bikeshedding on this one, if we really want to have a fixed ID that each site carries for anonymization purposes, please make it so that the value is randomly generated and stored in the database. A simple hash of the base URL is too easy to figure out because there are relatively few Open edX sites out there, and we can run SHA256 against all of them. Salting doesn't help because either the salt is randomly generated per-site (in which case it might as well be the anonymous value), or it's linked against some secret that might change.

We've walked up and down this path a couple of times with anonymous user IDs.

Reply:
@ormsbee I tough 2 difference options for this:

Setting: We can define a new setting to save the ID, the problem with this is that the ID could be changed by the user whenever he wants, so maybe is not the best idea.

Model: We can create a new model just to save the ID, this will help us to persist the ID without problems and the ID will never change, we can use a get_or_create in the send method.

What do you think?